### ISSUE TITLE
Allow custom names for WorkerMetricsStats and CoordinatorState tables

### ISSUE THREAD

**Issue by hittingray** [awslabs/amazon-kinesis-client#1410]
Hi there, wondering if it is possible for a custom table name to be passed in for the new WorkerMetricsStats and CoordinatorState tables. Obviously I can currently pass in an application name into `ConfigsBuilder`, which would then make the name `myApp-CoordinatorState`, but I would like to fully customise them to something else.

Thank you!

**Comment by hittingray** [comment]
Actually, looking a bit harder, it's possible via

```
        CoordinatorConfig coordinatorConfig = configsBuilder.coordinatorConfig();
        coordinatorConfig.coordinatorStateTableConfig().tableName("customNameGoesHere");

        LeaseManagementConfig leaseManagementConfig = configsBuilder.leaseManagementConfig();
        leaseManagementConfig.workerUtilizationAwareAssignmentConfig().workerMetricsTableConfig().tableName("otherCustomNameGoesHere");
```

Might be nice if it was a bit more obvious or documented better, since the lease table name is called out as a configuration option directly in the docs.

**Comment by akidambisrinivasan** [comment]
> Actually, looking a bit harder, it's possible via

Yes that is right.  The configurations are nested, so the same consistent set of configurations can be used for all tables. We will update the public docs to include a note for the customizations of the table configurations.

**Comment by minuhong-aws** [comment]
Hi @hittingray, thanks for bringing this up. Please note that you also need to set proper IAM permissions with these custom table names. 
https://docs.aws.amazon.com/streams/latest/dev/kcl-iam-permissions.html

Can you share the motivation of using custom table names for worker metrics and coordinator state tables? We also would like to hear about your use case and migration experience. You can email us via kinesis-kcl-feedback@amazon.com.

**Comment by dbottini2** [comment]
I'll comment from my own experience - I work at a large, microservice-driven organization. Our microservice are given least-permissions required to operate; and any resources required to run a service are created in parallel through CloudFormation or similar tools. As such, our overall architecture requires spinning up the table independently of the application, and only granting the service read/write access on the items of the table. At the very least, we would never be able to call CreateTable. And the resulting table names are subject to organization-driven naming semantics.

**Comment by madhuilluri** [comment]
Customizing `CoordinatorConfig` and `LeaseManagementConfig` to specify a new `tableName` for workers using KCL 3.x

```
LeaseManagementConfig leaseConfiguration = super.leaseManagementConfig();
leaseConfiguration
    .workerUtilizationAwareAssignmentConfig()
    .workerMetricsTableConfig()
    .tableName(kinesisWorkerMetricStatsTableName);


CoordinatorConfig coordinatorConfig = super.coordinatorConfig();
coordinatorConfig.coordinatorStateTableConfig().tableName(kinesisCoordinatorStateTableName);
```
